src/topology/builder.rs

```diff
@@ -562,3 +412,242 @@ const fn filter_event_type(event: &Event, data_type: DataType) -> bool {
         DataType::Metric => matches!(event, Event::Metric(_)),
     }
 }
+
+// 128 is an arbitrary, smallish constant
+const TRANSFORM_BATCH_SIZE: usize = 128;
```

Comment:
This is almost certainly too small when doing parallel transforms, as you have previously discussed.